client/components/plans-redesign/plan-actions/index.jsx

```diff
@@ -0,0 +1,285 @@
+					className="button is-primary plan-actions__upgrade-button"
+					disabled={ this.props.isSubmitting }
+					onClick={ this.handleSelectPlan }>
+					{ label }
+				</button>
+			</div>
+		);
+	},
+
+	recordUpgradeNowClick() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link', 'Product ID', this.props.plan.product_id );
+	},
+
+	recordUpgradeNowButton() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Button', 'Product ID', this.props.plan.product_id );
+	},
+
+	getCartItem( properties ) {
+		return cartItems.getItemForPlan( this.props.plan, properties );
+	},
+
+	handleSelectJetpackFreePlan( event ) {
+		event.preventDefault();
+
+		if ( this.props.onSelectFreeJetpackPlan ) {
+			return this.props.onSelectFreeJetpackPlan();
+		}
+	},
+
+	handleSelectPlan( event ) {
+		event.preventDefault();
+
+		if ( this.props.isSubmitting ) {
+			return;
+		}
+
+		if ( this.props.site && ! this.props.site.isUpgradeable() ) {
+			return;
+		}
+
+		const actionType = event.target.tagName === 'A' ? 'link' : 'button';
+
+		if ( 'link' === actionType ) {
+			this.recordUpgradeNowClick();
+		} else {
+			this.recordUpgradeNowButton();
+		}
+
+		const cartItem = isFreePlan( this.props.plan ) ? null : this.getCartItem();
+
+		if ( this.props.onSelectPlan ) {
+			return this.props.onSelectPlan( cartItem );
+		}
+		upgradesActions.addItem( cartItem );
+
+		const checkoutPath = this.props.selectedFeature && isValidFeatureKey( this.props.selectedFeature )
+			? `/checkout/features/${this.props.selectedFeature}/${ this.props.site.slug }`
+			: `/checkout/${ this.props.site.slug }`;
+
+		page( checkoutPath );
+	},
+
+	canSelectPlan() {
+		if ( ! config.isEnabled( 'upgrades/checkout' ) ) {
+			return false;
+		}
+
+		if ( this.props.site ) {
+			if ( this.siteHasThisPlan() ) {
+				return false;
+			}
+
+			return this.planHasCost() && ! isBusiness( this.props.site.plan );
+		}
+
+		return true;
+	},
+
+	getImageButton() {
+		const classes = classNames( 'plan-actions__illustration', this.props.plan.product_slug );
+
+		if ( ! this.canSelectPlan( this.props.plan ) ) {
+			return (
+				<div className={ classes } />
+			);
+		}
+
+		return (
+			<div onClick={ this.handleSelectPlan } className={ classes } />
+		);
+	},
+
+	downgradeMessage() {
+		return (
+			<small className="plan-actions__trial-period">{ this.translate( 'Contact support to downgrade your plan.' ) }</small>
+		);
+	},
+
+	siteHasThisPlan() {
+		return ! this.props.isInSignup && this.props.site && this.props.site.plan.product_id === this.props.plan.product_id;
+	},
+
+	managePlanButton() {
+		if ( this.planHasCost() && this.props.sitePlan.userIsOwner ) {
+			const link = purchasesPaths.managePurchase( this.props.site.slug, this.props.sitePlan.id );
+
+			return (
+				<a href={ link } className="button plan-actions__upgrade-button">{ this.translate( 'Manage Plan', { context: 'Link to current plan from /plans/' } ) }</a>
+			);
+		}
+	},
+
+	freePlanExpiration() {
```

Comment:
This function seems unnecessary, could just do the check and return directly since it's used once.